### PR TITLE
Fix false positive when TestCaseSource type is in another assembly

### LINKED ISSUE #304: Unable to compile test project when classlib with TestCaseSources is referenced
Hi,

I am trying to use NUnit.Analyzers and also TestCaseSource attribute documented on 

- https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcasesource.html
- https://docs.nunit.org/articles/nunit-analyzers/NUnit1011.html

I have prepared public gist showing the behavior I get while using classlib in test project solution. Maybe I am doing something wrong but I rather tell you about it as it might be scenario not used often.

Steps to reproduce:
- https://gist.github.com/adamvlasak/ff1acb7da54516b8f7a7625b339b88a5

Error I get (even though without NUnit.Analyzers tests run with no problem)
- https://gist.github.com/adamvlasak/ff1acb7da54516b8f7a7625b339b88a5#file-gistfile1-txt-L86

Can you please confirm it is a bug or tell me what I am doing wrong? Thank you very much.

### PR DESCRIPTION
**Pull request by Dreamescaper**
Fixes #304.

Analyzer cannot get source information when referenced type is in separate project.
Interestingly enough, in works in VS (no errors), but fails during build. 
Not sure why.

### CODE REVIEW

File: src/nunit.analyzers.tests/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzerTests.cs [hunk 1]
```diff
@@ -417,6 +418,51 @@ static IEnumerable<string> TestStrings(bool generateLongTestCase)
...
+            public static IEnumerable TestCaseDataProvider()
+            {
+                yield return new object[] { 1, 2, 3 };
+                yield return new object[] { 2, 3, 5 };
+            }
+        }
+    }";
+
+            var testDataReference = MetadataReferences.CreateBinary(testDataCode);
+            var references = AnalyzerAssert.MetadataReferences.Concat(new[] { testDataReference });
```
Comment (manfred-brands) [inline src/nunit.analyzers.tests/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzerTests.cs:457]:
```suggestion
            var references = AnalyzerAssert.MetadataReferences.Append(testDataReference);
```

**Review by manfred-brands — APPROVED** [review]
Slight optimization to use `Append(T)` instead of `Concat(new T[])`

File: src/nunit.analyzers.tests/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzerTests.cs (hunk 1 above)
Comment (Dreamescaper) [reply src/nunit.analyzers.tests/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzerTests.cs:457]: It's not available for net461, which is one of our test targets (I'm not sure why though)

**Review by mikkelbu — APPROVED** [review]
LGTM and thanks for correcting my wrong attempt at finding the symbol 👍 .

**Pull request merged**